File: classes/parte-8-como-funcionan-los-lenguajes/126-aot-vs-jit-costos-y-beneficios/implementaciones/c/main.c
#include <stdio.h>

int main(void) {
    int n;
    if (scanf("%d", &n) != 1) return 1;
    long r = 1;
    for (int i = 0; i < n; i++) r *= 2;
    printf("resultado=%ld\n", r);
    return 0;
}

File: classes/parte-8-como-funcionan-los-lenguajes/126-aot-vs-jit-costos-y-beneficios/implementaciones/rust/main.rs
use std::io::Read;

fn main() {
    let mut s = String::new();
    std::io::stdin().read_to_string(&mut s).unwrap();
    let n: u32 = s.trim().parse().unwrap();
    let r: i64 = 2i64.pow(n);
    println!("resultado={r}");
}
